### ISSUE TITLE
Nested lists do not adhere to WC3 guidelines and thus cannot be properly styled

### ISSUE THREAD

**Issue by toddpadwick** [sanity-io/sanity#4542]
When creating lists in Sanity block editor, all appears correct. however, when returned in portable text via GROQ, it seems that nested lists are not corrected nested in accordance with [WC3 guidelines](https://stackoverflow.com/questions/5899337/proper-way-to-make-html-nested-list).
HEre is an example: 
![Screenshot 2023-05-30 at 18 00 34](https://github.com/sanity-io/sanity/assets/36191462/477cbcdf-b692-4c78-aaa5-8e507bd9f3d9)
_It seems all looks correct in Sanitys editor (but on inspecting that seems because the editor does not use standard HTML to display it)._

## How it should be
The above editor content **_should_** return the following HTML structure:
```
    <ol>
         <li>Here is #1 top-level list item</li>
        <li>This is #2 top-level list item 
            <ol>         // note that this nested list is inside the second top level list item.
                <li>This is #1 second-level list item in #2 top-level list item.</li>
                <li>This is #2 second-level list item, still in #2 top-level list item</li>
            </ol>
        </li>        // note this is where the second top level list item is closed.
        <li>This is #3 top level list item</li>
    </ol>
```

## How it is
However, instead it renders this HTML, whereby the second level list item is inserted into a new, #3 list item in the top level list, pushing what should be #3 on the top level, into becoming #4.
```
    <ol>
         <li>Here is #1 top-level list item</li>
        <li>This is #2 top-level list item</li>
        <li> // note, a new list item is created, when it shouldn't be.
            <ol>
                <li>This is #1 second-level list item in #2 top-level list item.</li>
                <li>This is #2 second-level list item, still in #2 top-level list item</li>
            </ol>
        </li>
        <li>This is #3 top level list item</li>
    </ol>
```

This may seem trivial but in actual fact it causes the ordering of list items, to become completely out of whack:
![Screenshot 2023-05-30 at 18 06 58](https://github.com/sanity-io/sanity/assets/36191462/82789bb7-335c-4e87-a98e-39e3ae695181)

What should be list item three, is now list item 4, and the sub list that is relating to #2 is no longer associated with it, and has dropped to #3

**Comment by toddpadwick** [comment]
Anyone at Sanity able to have a glance at this Bug? Nesting lists may not be used often by many, but for one of our clients, it's an important necessity considering their blog articles are heavy on detailed technical specifications, with many instances of nested ordered lists.  (and please let me know if this should instead be reported on the [portable text repo](https://github.com/portabletext/portabletext) - happy to migrate it if that helps expedite the process)

**Comment by linkb15** [comment]
I dont face this issue anymore now - in latest sanity studio.

"@sanity/client": "^6.7.1",
"@portabletext/react": "^3.0.11",
"next-sanity": "5",
"sanity": "3",

**Comment by toddpadwick** [comment]
Thats good news @linkb15 . I will upgrade and try as soon as I get an opportunity to do so. thanks for letting me know :)

**Comment by linkb15** [comment]
yep no worries @toddpadwick  - first time integrating sanity cms in nextjs - trying out things myself as well.

I was searching how to customize ordered list to be like 1. 1.1. 1.1.1 ... and checking issues about list items here and there.

Then I was checking the internal code of the ordered list item as well and found out I can customize the component for the ordered list - will be quite tedious to customize the styling but should be possible.

**Comment by pedrobonamin** [comment]
Hello @toddpadwick thank you for reporting this.
I'm sorry it took us so long to get to you and reply here.

I believe this issue is now fixed.
This is rendering a list using '@portabletext/react' library https://github.com/portabletext/react-portabletext

<img width="632" height="508" alt="Image" src="https://github.com/user-attachments/assets/54ebbc26-da73-4bbd-9d5e-a9aecac0643b" />

The generated HTML is the following:

```html
<ol>
  <li>First item</li>
  <li>
    Second item
    <ol>
      <li>Nested 2.1</li>
      <li>Nested 2.2</li>
    </ol>
  </li>
  <li>Third item</li>
  <li>Fourth</li>
</ol>

```

Closing this, if you believe the issue persists please feel free to reopen it. Thank you!
